spec/models/reports/facility_state_spec.rb

```diff
@@ -67,4 +59,103 @@
+        ltfu_patient = create(:patient, recorded_at: june_2021[:over_12_months_ago], assigned_facility: facility)
+        create(:blood_pressure, patient: ltfu_patient, recorded_at: june_2021[:over_12_months_ago])
+        _different_facility_patient = create(:patient, recorded_at: june_2021[:long_ago])
+
+        RefreshMaterializedViews.new.refresh_v2
+        with_reporting_time_zone do
+          expect(described_class.find_by(month_date: june_2021[:now], facility: facility).under_care).to eq 1
+          expect(described_class.find_by(month_date: june_2021[:now], facility: facility).lost_to_follow_up).to eq 1
+          expect(described_class.find_by(month_date: june_2021[:now], facility: facility).dead).to eq 1
+          expect(described_class.find_by(month_date: june_2021[:now], facility: facility).cumulative_assigned_patients).to eq 3
+        end
+      end
+    end
+
+    describe "cumulative_assigned_patients" do
+      it "computes the total number of assigned patients as of the given month" do
+        facility = create(:facility)
+        create(:patient, assigned_facility: facility, recorded_at: june_2021[:now] - 2.years)
+        create(:patient, assigned_facility: facility, recorded_at: june_2021[:under_12_months_ago])
+        create(:patient, assigned_facility: facility, recorded_at: june_2021[:under_3_months_ago])
+        create(:patient, assigned_facility: facility, recorded_at: june_2021[:end_of_month])
+
+        RefreshMaterializedViews.new.refresh_v2
+        with_reporting_time_zone do
+          expect(described_class.find_by(facility: facility, month_date: june_2021[:now] - 2.years).cumulative_assigned_patients).to eq 1
+          expect(described_class.find_by(facility: facility, month_date: june_2021[:under_12_months_ago]).cumulative_assigned_patients).to eq 2
+          expect(described_class.find_by(facility: facility, month_date: june_2021[:under_3_months_ago]).cumulative_assigned_patients).to eq 3
+          expect(described_class.find_by(facility: facility, month_date: june_2021[:now]).cumulative_assigned_patients).to eq 4
+        end
+      end
+    end
+  end
+
+  context "treatment outcomes in the last 3 months" do
+    it "computes totals for under care patients" do
+      facility = create(:facility)
+
+      patient_controlled = create(:patient, assigned_facility: facility, recorded_at: june_2021[:long_ago])
+      create(:bp_with_encounter, :under_control, patient: patient_controlled, recorded_at: june_2021[:now] - 1.month)
+
+      patient_uncontrolled = create(:patient, assigned_facility: facility, recorded_at: june_2021[:long_ago])
+      create(:bp_with_encounter, :hypertensive, patient: patient_uncontrolled, recorded_at: june_2021[:now] - 1.months)
+
+      patient_missed_visit = create(:patient, assigned_facility: facility, recorded_at: june_2021[:long_ago])
+      create(:bp_with_encounter, patient: patient_missed_visit, recorded_at: june_2021[:over_3_months_ago])
```

Comment:
Want to add another missed visit patient, one who had no BPs or visits done at all?

Reply:
Done in 1735b493c47329439f21244bac76598aad456ebb. Thanks!